### ISSUE TITLE
Add support for IDataLoaderResult<>, new type from GraphQL 3.0.0

### ISSUE THREAD

**Issue by kuzn-ilya** [graphql-dotnet/conventions#215]
DataLoaders in GraphQL have been slightly changed in the 3.0.0 release. Now GraphQL expects IDataLoaderResult<> type, which has to be returned from resolvers instead of Task<>. But it seems like GraphQL.Conventions doesn't handle IDataLoaderResult<> properly.
For instance, this schema fails:
```
        private class BugReproQueryDataLoaderResult
        {
            public IDataLoaderResult<string> DataLoaderResult() => new DataLoaderResult<string>("Test");
        }
```
I believe, that it can be fixed similarly to #187, and I have a working solution, so I can raise a PR for this issue.

**Comment by tlil** [comment]
Cool - go for it :-)

**Comment by tlil** [comment]
Thank you, @kuzn-ilya  - merged in #216. Will follow up with a release tomorrow. :-)

**Comment by tlil** [comment]
@kuzn-ilya; now in https://github.com/graphql-dotnet/conventions/releases/tag/v3.3.2
